### PR TITLE
Add possibility to set the thread title

### LINKED ISSUE #50: ThreadTitle is always empty
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | Bundle 2.0.0; Sulu 2.5.10

#### Actual Behavior

Threads are created without a title.

#### Expected Behavior

Threads get the title of the page/article they belong to. 

#### Steps to Reproduce

Add comments form to article detail view and submit a comment -> a thread w/o a title is created.

#### Possible Solutions

Pass the Request-Param $threadTitle = $request->get('threadTitle') as an option to the comments-form when creating it inside \Sulu\Bundle\CommentBundle\Controller\WebsiteCommentController::cgetCommentsAction.
The rest can then be handled by extending the form-type and adding the intended title to the twig-render-call inside the article template.

### PR DESCRIPTION
**Pull request by Prokyonn**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes https://github.com/sulu/SuluCommentBundle/issues/50
| Related issues/PRs | #issuenum
| License | MIT
| Documentation PR | sulu/sulu-docs#prnum

#### What's in this PR?

Fix for https://github.com/sulu/SuluCommentBundle/issues/50

#### Example Usage

~~~php
    {{ render(path('sulu_comment.get_threads_comments', {
        threadId: 'page-' ~ uuid,
        referrer: app.request.uri,
        threadTitle: 'This is my title',
        _format: 'html'
    })) }}
~~~

#### To Do

- [ ] Create a documentation PR
- [ ] Update tests

### CODE REVIEW

File: Controller/WebsiteCommentController.php [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare(strict_types=1);
+
```
Comment (alexander-schranz) [inline Controller/WebsiteCommentController.php]: Adding strict_types is a bc break as params are not longer (cast) to correct type.

File: Twig/CommentFormFactoryTwigExtension.php [hunk 2]
```diff
@@ -52,6 +52,7 @@ public function createCommentForm(string $threadId, ?string $referrer = null, ?i
                 'threadId' => $threadId,
                 'referrer' => $referrer,
                 'parent' => $parent,
+                'threadTitle' => 'test',
```
Comment (alexander-schranz) [inline Twig/CommentFormFactoryTwigExtension.php]: This looks unexpected?

File: Controller/WebsiteCommentController.php [hunk 3]
```diff
@@ -186,14 +189,14 @@ public function cgetCommentsAction(string $threadId, Request $request): Response
                 'page' => $page ?: 1,
                 'pageSize' => $page ? $pageSize : null,
             ],
-            $this->getAdditionalContentData($request)
+            $this->getAdditionalContentData($request),
```
Comment (alexander-schranz) [inline Controller/WebsiteCommentController.php]: This probably will crash PHP 7.2 ?

File: Controller/WebsiteCommentController.php [hunk 4]
```diff
@@ -170,7 +172,8 @@ public function cgetCommentsAction(string $threadId, Request $request): Response
                 'data_class' => $this->commentClass,
                 'threadId' => $threadId,
                 'referrer' => $referrer,
-            ]
+                'threadTitle' => $request->get('threadTitle'),
```
Comment (alexander-schranz) [inline Controller/WebsiteCommentController.php]: we shoudl define where we want to get the threadTitle. If its a post it should be `$request->request->get('threadTitle')` if it GET `$request->query->...`

File: Controller/WebsiteCommentController.php (hunk 3 above)
Comment (Prokyonn) [reply Controller/WebsiteCommentController.php]: yeah did ran php-cs-fixer with 8.2 😬 

this is still a draft, nothing correctly tested yet 😅

File: Controller/WebsiteCommentController.php (hunk 4 above)
Comment (Prokyonn) [reply Controller/WebsiteCommentController.php]: fixed

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**